lib/src/migrators/module.dart

```diff
@@ -117,7 +138,86 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
     if (results == null) return null;
     var uses = _additionalUseRules
         .map((use) => '@use "$use"$_semicolonIfNotIndented\n');
-    return uses.join() + results;
+    return _getEntrypointForwards() + uses.join() + results;
+  }
+
+  /// Returns whether an member of [name] should be forwarded in the entrypoint.
+  ///
+  /// [name] should be the original name of that member, even if it started with
+  /// [prefixToRemove].
+  bool _shouldForward(String name) {
+    switch (forwardOption) {
+      case 'all':
+        return true;
+      case 'none':
+        return false;
+      case 'prefixed':
+        return name.startsWith(prefixToRemove);
+      default:
+        throw StateError('--forward should not allow invalid values');
+    }
+  }
+
+  /// If the current stylesheet is the entrypoint, return a string of @forward
+  /// rules to forward all members for which [_shouldForward] returns true.
+  String _getEntrypointForwards() {
+    if (!_scope.isGlobal) {
+      throw StateError('Must be called from root of stylesheet');
+    }
+    if (_upstreamStylesheets.isNotEmpty) return '';
+
+    // Divide all global members from dependencies into sets based on whether
+    // they should be forwarded or not.
+    var shown = <Uri, Set<String>>{};
+    var hidden = <Uri, Set<String>>{};
+    categorizeMember(Uri url, String originalName, String newName) {
+      if (url == _currentUrl) return;
+      if (_shouldForward(originalName)) {
+        shown[url] ??= {};
+        shown[url].add(newName);
+      } else {
+        hidden[url] ??= {};
+        hidden[url].add(newName);
+      }
+    }
+
+    for (var entry in _scope.variables.entries) {
+      if (entry.value is! VariableDeclaration) continue;
+      categorizeMember(entry.value.span.sourceUrl,
+          (entry.value as VariableDeclaration).name, '\$${entry.key}');
+    }
+    for (var entry in _scope.mixins.entries) {
+      categorizeMember(entry.value.span.sourceUrl, entry.value.name, entry.key);
+    }
+    for (var entry in _scope.functions.entries) {
+      categorizeMember(entry.value.span.sourceUrl, entry.value.name, entry.key);
+    }
+
+    // Create a @forward rule for each dependency that has members that should
+    // be forwarded.
+    var forwards = <String>[];
+    for (var url in shown.keys) {
+      var shownCount = shown[url].length;
+      var hiddenCount = hidden[url]?.length ?? 0;
+      var forward = '@forward "${_absoluteUriToDependency(url)}"';
+
+      // When not all members from a dependency should be forwarded, determine
+      // whether to use a `show` clause or a `hide` clause based on the number
+      // of members in each set.
```

Comment:
Nice! I like how this avoids `show`/`hide` declarations when they aren't necessary.